docs/cv-upload-feature/context.md

```diff
@@ -0,0 +1,256 @@
+- **packages/shared** - Shared components, hooks, and utilities
+- **packages/extension** - Browser extension (may need CV data integration)
+
+### Technology Stack
+- **React 18.3.1** with Next.js 15.4.1
+- **TypeScript** for type safety
+- **TanStack Query v5** for server state management
+- **GraphQL** with graphql-request for API communication
+- **Tailwind CSS** with custom design system
+- **Jotai** for local state management
+
+### File Upload Implementation Requirements
+
+#### File Handling
+- **Supported formats**: PDF (primary), DOC/DOCX likely
+- **File size limits**: TBD (typical CV size constraints)
+- **Validation**: File type, size, and basic structure validation
+- **Storage**: Backend integration for secure file storage
+- **Processing**: CV parsing/analysis capabilities (future)
+
+#### Drag & Drop Implementation
+- Native HTML5 drag-and-drop API
+- Visual feedback during drag operations
+- Error handling for unsupported files
+- Progress indicators during upload
+
+### API Integration
+
+#### GraphQL Operations Needed
+```typescript
+// Upload CV mutation
+mutation UploadCV($file: Upload!, $userId: ID!) {
+  uploadCV(file: $file, userId: $userId) {
+    id
+    filename
+    uploadedAt
+    status
+  }
+}
+
+// Get user CV query  
+query GetUserCV($userId: ID!) {
+  user(id: $userId) {
+    cv {
+      id
+      filename
+      uploadedAt
+      status
+    }
+  }
+}
+
+// Delete/replace CV mutation
+mutation DeleteCV($cvId: ID!) {
+  deleteCV(id: $cvId) {
+    success
+  }
+}
+```
+
+#### Analytics Integration
+- New event: `upload cv` (as specified in PRD)
+- Track upload attempts, successes, failures
+- Measure banner dismissal rates
+- Monitor LinkedIn import usage
+
+### Component Architecture
+
+#### Shared Components to Create/Extend
+Based on existing design system patterns:
+
+1. **CVUploadBanner** (new) - Homepage feed banner
+   - Location: `packages/shared/src/components/cards/`
+   - Drag & drop zone with visual feedback
+   - LinkedIn import link integration
+
+2. **CVUploadSection** (new) - Profile page upload area
+   - Location: `packages/shared/src/components/profile/`
+   - File management interface
+   - Upload status display
+
+3. **CVUploadModal** (new) - Success/error modals
+   - Location: `packages/shared/src/components/modals/`
+   - Celebration modal for successful uploads
+   - Error handling modal for failed uploads
+
+4. **FileUploadField** (new) - Reusable drag-and-drop field
+   - Location: `packages/shared/src/components/fields/`
+   - Generic file upload with customizable validation
+
+#### Existing Components to Leverage
+- **Button** - For browse files, LinkedIn link buttons
+- **Toast** - For dismissal feedback messages
+- **ContentModal** - Base for upload success/error modals
+- **Typography** - Consistent text styling throughout
+
+### State Management Strategy
+
+#### Local State (Jotai)
+- Upload progress tracking
+- Drag-and-drop state
```

Comment:
Use simple local state with useState/useReducer hook, upload/d&d does not need jotai